m2cgen/interpreters/interpreter.py

```diff
@@ -29,6 +29,9 @@ def _do_interpret(self, expr, to_reuse=None, **kwargs):
         if result is not None:
             return result
 
+        if expr in self._cached_expr_results:
+            return self._cached_expr_results[expr].var_name
```

Comment:
Search in cache for expressions that are now not `to_reuse` but might be cached earlier.